src/views/clusteroverview/overview/components/details-card/hooks/useKubevirtCSVDetails.ts

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
+
+import CatalogSourceModel from '@kubevirt-ui/kubevirt-api/console/models/CatalogSourceModel';
+import ClusterServiceVersionModel from '@kubevirt-ui/kubevirt-api/console/models/ClusterServiceVersionModel';
+import SubscriptionModel from '@kubevirt-ui/kubevirt-api/console/models/SubscriptionModel';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { K8sResourceCommon, useK8sWatchResources } from '@openshift-console/dynamic-plugin-sdk';
+import { useDeepCompareMemoize } from '@openshift-console/dynamic-plugin-sdk/lib/utils/k8s/hooks/useDeepCompareMemoize';
+
+import { CatalogSourceKind, ClusterServiceVersionKind, SubscriptionKind } from '../utils/types';
+
+import { useDebounceCallback } from './useDebounceCallback';
+
+const getKubevirtCSV = (clusterServiceVersions, installedCSV): ClusterServiceVersionKind => {
+  return clusterServiceVersions.find((csv) => csv?.metadata?.name === installedCSV);
+};
+
+const getKubevirtSubscription = (subscriptions) => {
+  return subscriptions.find((sub) => sub?.metadata?.name === 'hco-operatorhub');
+};
+
+const isPackageServer = (obj) =>
+  obj?.metadata?.name === 'packageserver' &&
+  obj?.metadata?.namespace === 'openshift-operator-lifecycle-manager';
+
+const catalogSourceForSubscription = (
+  subscription: SubscriptionKind,
+  catalogSources: CatalogSourceKind[] = [],
+): CatalogSourceKind =>
+  catalogSources.find(
+    (source) =>
+      source?.metadata?.name === subscription?.spec?.source &&
+      source?.metadata?.namespace === subscription?.spec?.sourceNamespace,
+  );
+
+type UseKubevirtCSVDetails = {
+  name: string;
+  provider: string;
+  version: string;
+  updateChannel: string;
+  kubevirtSub: SubscriptionKind;
+  catalogSourceMissing: boolean;
+  loaded: boolean;
+  loadError: string;
+};
+
+export const useKubevirtCSVDetails = (): UseKubevirtCSVDetails => {
+  const [name, setName] = React.useState<string>();
+  const [provider, setProvider] = React.useState<string>();
+  const [version, setVersion] = React.useState<string>();
+  const [updateChannel, setUpdateChannel] = React.useState<string>();
+  const [kubevirtSub, setKubevirtSub] = React.useState<SubscriptionKind>();
+  const [catalogSourceMissing, setCatalogSourceMissing] = React.useState<boolean>(false);
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [loadError, setLoadError] = React.useState<string>('');
+
+  const watchedResources = {
+    installedCSVs: {
+      groupVersionKind: {
```

Comment:
would be nicer to do `groupVersionKind: modelToGroupVersionKind(ClusterServiceVersionModel)`
or maybe simply import `ClusterServiceVersionModelGroupVersionKind` :)

Reply:
Thanks! I forgot about those helpers. Fixed!